### ISSUE TITLE
kafka install failure: Unknown task kind Toggle

### ISSUE THREAD

**Issue by t0gre** [kudobuilder/frameworks#260]
To replicate: 

Literally follow the [runbook](https://kudo.dev/docs/runbooks/kafka/install.html#installing-the-operator) exactly

1. Fresh EKS cluster 
2. `k kudo init`. 
3. `k kudo install zookeeper`
4. `k kudo install kafka`

Result = install fails.

`k kudo plan status --instance=kakfa-instance` shows:

```
Plan(s) for "kafka-instance" in namespace "default":
.
└── kafka-instance (Operator-Version: "kafka-1.3.0" Active-Plan: "deploy")
    ├── Plan cruise-control (serial strategy) [NOT ACTIVE]
    │   └── Phase cruise-addon (serial strategy) [NOT ACTIVE]
    │       └── Step deploy-cruise-control [NOT ACTIVE]
    ├── Plan deploy (serial strategy) [FATAL_ERROR]
    │   ├── Phase deploy-kafka (serial strategy) [COMPLETE]
    │   │   ├── Step generate-tls-certificates [COMPLETE]
    │   │   ├── Step configuration [COMPLETE]
    │   │   ├── Step service [COMPLETE]
    │   │   └── Step app [COMPLETE]
    │   └── Phase addons (parallel strategy) [FATAL_ERROR]
    │       ├── Step monitoring [FATAL_ERROR] (default/kafka-instance fatal error:  failed to build task deploy.addons.monitoring.service-monitor: unknown task kind Toggle)
    │       ├── Step access [PENDING]
    │       ├── Step mirror [PENDING]
    │       └── Step load [PENDING]
    ├── Plan external-access (serial strategy) [NOT ACTIVE]
    │   └── Phase resources (serial strategy) [NOT ACTIVE]
    │       └── Step deploy [NOT ACTIVE]
    ├── Plan kafka-connect (serial strategy) [NOT ACTIVE]
    │   └── Phase deploy-kafka-connect (serial strategy) [NOT ACTIVE]
    │       ├── Step deploy [NOT ACTIVE]
    │       └── Step setup [NOT ACTIVE]
    ├── Plan mirrormaker (serial strategy) [NOT ACTIVE]
    │   └── Phase app (serial strategy) [NOT ACTIVE]
    │       └── Step deploy [NOT ACTIVE]
    ├── Plan not-allowed (serial strategy) [NOT ACTIVE]
    │   └── Phase not-allowed (serial strategy) [NOT ACTIVE]
    │       └── Step not-allowed [NOT ACTIVE]
    ├── Plan service-monitor (serial strategy) [NOT ACTIVE]
    │   └── Phase enable-service-monitor (serial strategy) [NOT ACTIVE]
    │       └── Step deploy [NOT ACTIVE]
    ├── Plan update-instance (serial strategy) [NOT ACTIVE]
    │   └── Phase app (serial strategy) [NOT ACTIVE]
    │       ├── Step conf [NOT ACTIVE]
    │       ├── Step svc [NOT ACTIVE]
    │       └── Step sts [NOT ACTIVE]
    └── Plan user-workload (serial strategy) [NOT ACTIVE]
        └── Phase workload (serial strategy) [NOT ACTIVE]
            └── Step toggle-workload [NOT ACTIVE]
```
 Output of `k kudo version` is 

```
KUDO Version: version.Info{GitVersion:"0.12.0", GitCommit:"50a43f45", BuildDate:"2020-04-15T19:26:34Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by porridge** [comment]
@tomgreenwood1 please show `k get -n kudo statefulset kudo-controller-manager -o yaml`

**Comment by t0gre** [comment]
@porridge `k get -n kudo statefulset kudo-controller-manager -o yaml` returns `Error from server (NotFound): namespaces "kudo" not found` but `k get -n kudo-system statefulset kudo-controller-manager -o yaml` returns: 

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  creationTimestamp: "2020-05-12T07:37:00Z"
  generation: 1
  labels:
    app: kudo-manager
    control-plane: controller-manager
  name: kudo-controller-manager
  namespace: kudo-system
  resourceVersion: "1982739"
  selfLink: /apis/apps/v1/namespaces/kudo-system/statefulsets/kudo-controller-manager
  uid: 5da8e61f-9423-11ea-a11b-0a137a82cb70
spec:
  podManagementPolicy: OrderedReady
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: kudo-manager
      control-plane: controller-manager
  serviceName: kudo-controller-manager-service
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: kudo-manager
        control-plane: controller-manager
    spec:
      containers:
      - command:
        - /root/manager
        env:
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: SECRET_NAME
          value: kudo-webhook-server-secret
        - name: ENABLE_WEBHOOKS
          value: "false"
        image: kudobuilder/controller:v0.10.0
        imagePullPolicy: Always
        name: manager
        ports:
        - containerPort: 443
          name: webhook-server
          protocol: TCP
        resources:
          requests:
            cpu: 100m
            memory: 50Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: kudo-manager
      serviceAccountName: kudo-manager
      terminationGracePeriodSeconds: 10
  updateStrategy:
    rollingUpdate:
      partition: 0
    type: RollingUpdate
status:
  collisionCount: 0
  currentReplicas: 1
  currentRevision: kudo-controller-manager-56cf8595f8
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updateRevision: kudo-controller-manager-56cf8595f8
  updatedReplicas: 1
```
Thanks

**Comment by porridge** [comment]
You have an old controller: `image: kudobuilder/controller:v0.10.0` please upgrade.

**Comment by porridge** [comment]
cc @ANeumann82 who worked on upgrades IIRC

**Comment by t0gre** [comment]
ok thanks. What is the recommended version? From dockerhub it looks like the latest is v0.13.0-rc1...  also, I ran `k kudo init` this morning, seems strange it pulled an old version?

**Comment by porridge** [comment]
`rc1` is pre-release. I'd recommend using https://github.com/kudobuilder/kudo/releases/latest

I'm not sure how `init` works when there is already a controller installed there :thinking:

**Comment by porridge** [comment]
OTOH `creationTimestamp: "2020-05-12T07:37:00Z"` sounds like the resource was indeed created recently :thinking: 
This is... weird.

**Comment by porridge** [comment]
How did you obtain the `kudo` plugin, @tomgreenwood1 ?

**Comment by ANeumann82** [comment]
This is pretty weird. The `k kudo version` output indicates that the installed controller should be 0.12.0 as well...

**Comment by porridge** [comment]
@tomgreenwood1 the only explanation I can come up with is that somehow you ran a different `kudo` plugin version when you ran `init`.
That or some weird bug in our CLI. Can you reproduce this with another fresh EKS cluster?

**Comment by zmalik** [comment]
@tomgreenwood1 were you able to test it again?

**Comment by t0gre** [comment]
Hi All, 

Sorry for the slow reply. I've been busy with work but just did a fresh install now and everything worked fine. Plan status is healthy, `k kudo init` installed `v0.12.0`.. everything went fine basically, and I haven't upgraded the kudo plugin (@porridge I got the plugin from krew). So I dunno, maybe I did something wrong last time, but I can't figure out what.. but yeah, I can't reproduce it now, so maybe just call it resolved? 

Thanks for the help
